sparkle/src/components/ConfettiBackground.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef, useState } from "react";
+import Confetti from "react-confetti";
+
+function randomInt(min: number, max: number): number {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function drawSnowflake(ctx: CanvasRenderingContext2D): void {
+  const numPoints = randomInt(3, 4) * 2;
+  const innerRadius = 0.2;
+  const outerRadius = 0.8;
+  ctx.beginPath();
+  ctx.moveTo(0, 0 - outerRadius);
+
+  for (let n = 1; n < numPoints * 2; n++) {
+    const radius = n % 2 === 0 ? outerRadius : innerRadius;
+    const x = radius * Math.sin((n * Math.PI) / numPoints);
+    const y = -1 * radius * Math.cos((n * Math.PI) / numPoints);
+    ctx.lineTo(x, y);
+  }
+  ctx.fill();
+  ctx.stroke();
+  ctx.closePath();
+}
+
+export interface ConfettiBackgroundProps {
+  width?: number;
+  height?: number;
+  variant?: "confetti" | "snow";
+  referentSize?: React.RefObject<HTMLElement>;
+}
+
+const ConfettiBackground: React.FC<ConfettiBackgroundProps> = ({
+  width,
+  height,
+  variant = "confetti",
+  referentSize,
+}) => {
+  const [referentWidth, setReferentWidth] = useState(0);
+  const [referentHeight, setReferentHeight] = useState(0);
+  const resizeObserver = useRef<ResizeObserver | null>(null);
+
+  useEffect(() => {
+    if (referentSize && referentSize.current) {
+      const { clientWidth, clientHeight } = referentSize.current;
+      setReferentWidth(clientWidth);
+      setReferentHeight(clientHeight);
+
+      resizeObserver.current = new ResizeObserver((entries) => {
+        for (const entry of entries) {
+          setReferentWidth(entry.target.clientWidth);
+          setReferentHeight(entry.target.clientHeight);
+        }
+      });
+
+      resizeObserver.current.observe(referentSize.current);
+    }
+
+    return () => {
+      if (resizeObserver.current && referentSize && referentSize.current) {
+        resizeObserver.current.unobserve(referentSize.current);
+      }
+    };
+  }, [referentSize]);
+
+  const confettiProps = {
```

Comment:
We should not define this in the component, as will be recreated every time, let's move it above the component and only add the width and height in the switch/case below.